action.yml

```diff
@@ -19,13 +22,31 @@ runs:
           actionlint ${{ inputs.actionlint-version }}
 
     - name: Configure actionlint
-      shell: bash
+      shell: python
       run: |
-        cat << EOF > .github/actionlint.yaml
-        self-hosted-runner:
-          labels:
-            - aws-gpu-runner
-        EOF
+        import yaml
+        from pathlib import Path
+
+        config_path = Path(".github/actionlint.yaml")
+
+        def get_config():
+            if not config_path.is_file():
+                return {}
+            with config_path.open() as fd:
+                return yaml.safe_load(fd)
+
+        def save_config(config):
+            with config_path.open("w") as fd:
+                yaml.safe_dump(config, fd)
+
+        config = get_config()
+        labels = "${{ inputs.self-hosted-runners }}".split(",")
+        config.setdefault("self-hosted-runner", {}).setdefault("labels", []).extend(labels)
```

Comment:
Yay, so Pythonic. I love it.

Now, how does ActionLint react if `self-hosted-runner.labels` contains multiple times the same entry?

Reply:
deduplicated labels